tests/unit/amo/components/TestSearchForm.js

```diff
@@ -219,6 +219,173 @@ describe(__filename, () => {
       }));
     });
 
+    it('does not fetch suggestions twice on focus when search has not changed', () => {
+      const { store } = dispatchAutocompleteResults({ results: [
+        createFakeAutocompleteResult(),
+        createFakeAutocompleteResult(),
+      ] });
+      const { autocomplete: autocompleteState } = store.getState();
+
+      const dispatch = sinon.spy();
+      const wrapper = mountComponent({
+        dispatch,
+        suggestions: autocompleteState.suggestions,
+      });
+
+      sinon.assert.notCalled(dispatch);
+
+      // User types 'a' in the search input.
+      wrapper.find('input').simulate('change', createFakeChangeEvent('a'));
+
+      // User clicks somewhere else on the UI, triggering the
+      // AUTOCOMPLETE_CANCELLED action.
+      store.dispatch(autocompleteCancel());
+      wrapper.setProps(mapStateToProps(store.getState()));
+
+      // User focuses the search input.
+      wrapper.find('input').simulate('focus');
+
+      expect(wrapper.find(Suggestion)).toHaveLength(
+        autocompleteState.suggestions.length
+      );
+
+      sinon.assert.callCount(dispatch, 1);
+      sinon.assert.calledWith(dispatch, autocompleteStart({
+        errorHandlerId: errorHandler.id,
+        filters: { query: 'a' },
+      }));
+    });
+
+    it('fetches suggestions twice on focus when search has not changed but add-type changed', () => {
+      const { store } = dispatchAutocompleteResults({ results: [
+        createFakeAutocompleteResult(),
+        createFakeAutocompleteResult(),
+      ] });
+      const { autocomplete: autocompleteState } = store.getState();
+
+      const dispatch = sinon.spy();
+      const wrapper = mountComponent({
+        dispatch,
+        suggestions: autocompleteState.suggestions,
+      });
+
+      sinon.assert.notCalled(dispatch);
+
+      // User types 'a' in the search input.
+      wrapper.find('input').simulate('change', createFakeChangeEvent('a'));
+
+      // User clicks somewhere else on the UI, triggering the
+      // AUTOCOMPLETE_CANCELLED action.
+      store.dispatch(autocompleteCancel());
+      wrapper.setProps(mapStateToProps(store.getState()));
+      wrapper.setProps({ addonType: ADDON_TYPE_THEME });
+
+      // User focuses the search input.
+      wrapper.find('input').simulate('focus');
+
+      expect(wrapper.find(Suggestion)).toHaveLength(
+        autocompleteState.suggestions.length
+      );
+
+      sinon.assert.callCount(dispatch, 2);
+      sinon.assert.calledWith(dispatch, autocompleteStart({
```

Comment:
I think this test would be ever more robust if you reset the dispatch before `wrapper.setProps({ addonType: ADDON_TYPE_THEME });`. Technically this isn't checking the call order, which I know is minor but if you instead made sure it's called once after each event we expect the dispatch on we can check the order too.

It's super minor but I think it's better overall.